### ISSUE TITLE
Understanding of vpred and values

### ISSUE THREAD

**Issue by MrToser** [volcengine/verl#2642]
In the compute_value_loss function, aren't vpred and values both derived from the same batch and generated through _forward_micro_batch? 

Theoretically speaking, these two values should be the same. Am I understanding this correctly?

**Comment by wuxibin89** [comment]
Critic model is trained in `ppo_mini_batch_size`,  for example if `train_batch_size=1024,ppo_mini_batch_size=256`, then in a single PPO step, critic model is updated 4 times(1024/256).
- `vpred`: values predict by critic model before this mini step.
- `values`: values predict by critic model before this ppo step.

**Comment by MrToser** [comment]
At the first mini step of critic in each train_batch_size, the old and new networks are the same.  At this point, the value and vpred should be the same.

In this case, loss can be simplified as：loss=(y-y.detach+advtages)^2
Does the loss have much significance? Could it be reduced to on-policy?

I'm not sure if it would be more reasonable to set the losses separately.

**Comment by wuxibin89** [comment]
The `values` is only used for `vpreds` clip, in first mini step, the `values` and `vpreds` are same, so the clip is a no-op.
https://github.com/volcengine/verl/blob/main/verl/trainer/ppo/core_algos.py#L1025

**Comment by MrToser** [comment]
Let's examine the calculation method of loss. For now, we won't consider the clip operation, as the clip is merely an optimization and does not affect the essence.

Then the entire expression becomes

vf_loss = 1/2 * (vf_pred - return)^2

The calculation of "return" is carried out under the condition of "with torch.no_grad()", and the gradients are not involved.

return=advantages+value

Then vf_loss becomes

vf_loss = 1/2 * (vf_pred - advantages - value)^2

Here, only vf_pred is parameterized, while the other items are equivalent to constants. However, since vf_pred and value are equal in this case, the significance of this optimization seems to be less clear?

**Comment by MrToser** [comment]
I'm thinking that at this point, we could simply revert to — setting the "advantage" to 0 and including the advantage in the gradient calculation.

I don't know what will happen with this separation. It's just a question I came across during my learning process, because the first batch had nothing to do with the old and new networks.

**Comment by Kirrito-k423** [comment]
### 中文
定时清理检查发现：此 issue 最近一次非当前运行账号的公开回复或引用时间是 `2025-07-22T07:37:17Z`（comment，来自 `MrToser`），相对于本轮运行日期 `2026-07-12` 已超过 6 个月。因此按社区 issue 定时清理标准关闭。

已检查证据：Latest non-self public reply/reference: `2025-07-22T07:37:17Z` (comment by `MrToser`).
Comments API: https://api.github.com/repos/verl-project/verl/issues/2642/comments
Timeline API: https://api.github.com/repos/verl-project/verl/issues/2642/timeline

如果作者或社区成员仍能提供新的复现、证据或需求背景，请在此 issue 下继续评论请求重开，或创建新的 issue 并链接到这里。

### English
Scheduled cleanup check found that the latest public reply or reference by an account other than the current automation account was `2025-07-22T07:37:17Z` (comment by `MrToser`), which is more than 6 months before this run date `2026-07-12`. Therefore this issue is being closed under the scheduled stale-issue cleanup rule.

Evidence checked: Latest non-self public reply/reference: `2025-07-22T07:37:17Z` (comment by `MrToser`).
Comments API: https://api.github.com/repos/verl-project/verl/issues/2642/comments
Timeline API: https://api.github.com/repos/verl-project/verl/issues/2642/timeline

If the author or community members can still provide a fresh reproduction, evidence, or requirement context, please comment here to request reopening, or open a new issue and link back to this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
